components/list/demo/list-drag-n-drop-sample.js

```diff
@@ -0,0 +1,78 @@
+import '../list-item-content.js';
+import './list-item-sample.js';
+import '../list.js';
+import { html, LitElement } from 'lit-element/lit-element.js';
+import { repeat } from 'lit-html/directives/repeat';
+
+class listDragNDropSample extends LitElement {
+	static get properties() {
+		return {
+			list: { type: Array }
+		};
+	}
+
+	constructor() {
+		super();
+		this.list = [
+			{
+				key: '1',
+				name: 'Geomorphology and GIS',
+				secondary: 'This course explores the geological processes of the Earth\'s interior and surface. These include volcanism, earthquakes, mountain...',
+				img: 'https://s.brightspace.com/course-images/images/63b162ab-b582-4bf9-8c1d-1dad04714121/tile-high-density-max-size.jpg'
+			},
+			{
+				key: '2',
+				name: 'Engineering Materials for Energy Systems',
+				secondary: 'This course explores the geological processes of the Earth\'s interior and surface. These include volcanism, earthquakes, mountain...',
+				img: 'https://s.brightspace.com/course-images/images/e5fd575a-bc14-4a80-89e1-46f349a76178/tile-high-density-max-size.jpg'
+			},
+			{
+				key: '3',
+				name: 'Introductory Earth Sciences',
+				secondary: 'This course explores the geological process of the Earth\'s interior and surface. These include volcanism, earthquakes, mountains...',
+				img: 'https://s.brightspace.com/course-images/images/38e839b1-37fa-470c-8830-b189ce4ae134/tile-high-density-max-size.jpg'
+			},
+			{
+				key: '4',
+				name: 'Applied Complex Analysis',
+				secondary: 'This course explores the geological process of the Earth\'s interior and surface. These include volcanism, earthquakes, mountains...',
+				img: 'https://s.brightspace.com/course-images/images/c63e7407-c3ba-4fa0-8383-08a8f4fa468b/tile-high-density-max-size.jpg'
+			},
+			{
+				key: '5',
+				name: 'Basic French',
+				secondary: 'This course explores the geological process of the Earth\'s interior and surface. These include volcanism, earthquakes, mountains...',
+				img: 'https://s.brightspace.com/course-images/images/9e319eb4-31af-4912-889d-92d9f2d82884/tile-high-density-max-size.jpg'
+			},
+			{
+				key: '6',
+				name: 'Algebraic Number Theory',
+				secondary: 'This course explores the geological process of the Earth\'s interior and surface. These include volcanism, earthquakes, mountains...',
+				img: 'https://s.brightspace.com/course-images/images/36c5813d-2ac3-4a73-8f39-3d6e1b381fe3/tile-high-density-max-size.jpg'
+			}
+		];
+	}
+
+	render() {
+		return html`
+			<d2l-list>
+				${repeat(this.list, (item) => item.key, (item) => html`
```

Comment:
If you use keyboard mode and move an item up. It works!! Move an item down and we lose focus and thus keyboard mode.

My solution to this is making a better directive. If anyone has something better let me know! Not sure how much time we have to create the directive :O

Reply:
Let's get to the bottom of that - sounds like focus is being moved elsewhere for some reason.  Maybe worst case scenario is we have to reapply focus after it's been moved.

Reply:
Yeah, this is a tricking one. So in the prototype the issue was that the object moving got re-render when moving down. I believe it has to do with how repeat handles its slices. 

I thought about doing the focus after it has been moved. What makes that hard and out of my reach is the blur happens at this level and so the list items don't really know when it is done. I would love a solution where the person using the drag n drop list doesn't need to know about it. I really like how simple it is to actually use drag n drop list.